tarofreighter/parcel.go

```diff
@@ -200,7 +200,44 @@ type sendPackage struct {
 // anchor output as well as the Taro script root of the output (the Taproot
 // tweak).
 func (s *sendPackage) inputAnchorPkScript() ([]byte, []byte, error) {
-	taroScriptRoot := s.InputAsset.Commitment.TapscriptRoot(nil)
+	// If the input asset was received non-interactively, then the Taro tree
+	// of the input anchor output was built with asset leaves that had empty
+	// SplitCommitments. However, the SplitCommitment field was
+	// populated when the transfer of the input asset was verified.
+	// To recompute the correct output script, we need to build a Taro tree
+	// from the input asset without any SplitCommitment.
+
+	inputAssetCopy := s.InputAsset.Asset.Copy()
+	inputAnchorCommitmentCopy, err := s.InputAsset.Commitment.Copy()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Assets received via non-interactive split should have one witness,
+	// with an empty PrevID and a SplitCommitment present.
+	if len(inputAssetCopy.PrevWitnesses) == 1 &&
+		*inputAssetCopy.PrevWitnesses[0].PrevID == asset.ZeroPrevID &&
+		inputAssetCopy.PrevWitnesses[0].SplitCommitment != nil {
+
+		inputAssetCopy.PrevWitnesses[0].SplitCommitment = nil
+
+		// Build the new Taro tree by first updating the asset
+		// commitment tree with the new asset leaf, and then the
+		// top-level Taro tree.
+		inputAssetTree := inputAnchorCommitmentCopy.
```

Comment:
I think we can use an intermediate variable here (or two) to get slightly prettier line wrapping?